components/wmtk_components/input/internal/mesh_with_tag_from_image.cpp

```diff
@@ -0,0 +1,80 @@
+#include "mesh_with_tag_from_image.hpp"
+
+#include <wmtk/utils/mesh_utils.hpp>
+
+#define STB_IMAGE_IMPLEMENTATION
+#include <stb_image.h>
+
+namespace wmtk::components::internal {
+
+Eigen::Matrix<long, -1, -1> tags_from_image(const std::filesystem::path& file)
+{
+    int width, height, channels;
+    unsigned char* img = stbi_load(file.string().c_str(), &width, &height, &channels, 0);
+    if (img == NULL) {
+        throw std::runtime_error("Error in loading the image");
+    }
+    const size_t img_size = width * height * channels;
+
+    Eigen::Matrix<long, -1, -1> pixel_matrix;
+    pixel_matrix.resize(height, width);
+
+    for (size_t i = 0; i < height; ++i) {
+        for (size_t j = 0; j < width; ++j) {
+            const int pixel = *(img + ((width * i) + j) * channels);
+            pixel_matrix(i, j) = pixel;
+        }
+    }
+
+    return pixel_matrix;
+}
+
+std::shared_ptr<wmtk::TriMesh> mesh_with_tag_from_image(
+    const std::filesystem::path& file,
+    const std::string& tag_name)
+{
+    std::shared_ptr<wmtk::TriMesh> m = std::make_shared<wmtk::TriMesh>();
+
+    Eigen::Matrix<long, -1, -1> img = tags_from_image(file);
+
+    std::cout << "img:\n" << img << std::endl;
+
+    auto lex_index_v = [&img](size_t i, size_t j) { return i * (img.cols() + 1) + j; };
```

Comment:
it's eigen :( so everything should be int (cuz img.cols() returns an int. surprised you didn't get narrowing warnings and whatnot everywhere